lib/modules/ens/embarkjs.js

```diff
@@ -245,5 +245,5 @@ __embarkENS.registerSubDomain = function (name, address, callback) {
 };
 
 __embarkENS.isAvailable = function () {
-  return Boolean(this.isAvailable);
+  return Boolean(this._isAvailable);
```

Comment:
That was always returning true at first because: Boolean(Func)
Then once the set provider finish it was failing because it is not a function anymore because of:
`this.isAvailable = true`